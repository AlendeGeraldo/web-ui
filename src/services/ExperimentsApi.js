// CORE LIBS
import axios from 'axios';

// CONSTANTS
// api base url
const URL = process.env.REACT_APP_PROJECTS_API || 'http://localhost:3000';
// projects path
const projectsPath = '/projects';
// api object
const experimentsApi = axios.create({
  baseURL: `${URL}${projectsPath}`,
});
// experiments path
const experimentsPath = '/experiments';

// API METHODS
/**
 * List Experiments
 * @param {string} projectId
 * @returns {Promise}
 */
const listExperiments = (projectId) =>
  new Promise((resolve, reject) => {
    // requesting projects
    experimentsApi
      .get(`/${projectId}${experimentsPath}`)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  });

// EXPORT DEFAULT
export default {
  listExperiments,
};

// export const getExperimentList = async (id) => {
//   try {
//     const response = await projectsApi.get(`/projects/${id}/experiments`);
//     return response;
//   } catch (error) {
//     message.error(error.message);
//   }
// };

// export const getExperiment = async (projectId, experimentId) => {
//   try {
//     const response = await projectsApi.get(
//       `/projects/${projectId}/experiments/${experimentId}`
//     );
//     return response;
//   } catch (error) {
//     message.error(error.message);
//   }
// };

// export const createExperiment = async (id, experimentName) => {
//   try {
//     const body = {
//       name: experimentName,
//     };
//     const response = await projectsApi.post(
//       `/projects/${id}/experiments/`,
//       body
//     );
//     return response;
//   } catch (error) {
//     message.error(error.message);
//   }
// };

// export const updateExperiment = async (projectId, experimentId, body) => {
//   try {
//     const response = await projectsApi.patch(
//       `/projects/${projectId}/experiments/${experimentId}`,
//       body
//     );
//     return response;
//   } catch (error) {
//     if (error.response) message.error(error.response.data.message);
//   }
// };

// // \"name\": \"Auto-featuring Experiment\",
// // \"pipelineIdTrain\": \"23266cfd-4ed6-43d6-b8a0-ca8440d251c6\",
// // \"pipelineIdDeploy\": \"fe5205f5-7f76-4f57-84ca-ea6dd62670e8\",
// // \"datasetId\": \"0a10c0ac-ff3b-42df-ab7a-dc2962a1750c\",
// // \"headerId\": \"482b603f-23c1-4a10-9b79-8c5b91c6c0cb\",
// // \"targetColumnId\": \"3191a035-97a6-4e29-90d4-034cb1f87237\",
// // \"parameters\": \"{ price: 8, auto-featuring: true }\",
// // \"position\": 1
