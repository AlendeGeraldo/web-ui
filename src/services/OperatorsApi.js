// CORE LIBS
import axios from 'axios';

// CONSTANTS
// api base url
const URL = process.env.REACT_APP_PROJECTS_API || 'http://localhost:3000';
// projects path
const projectsPath = '/projects';
// api object
const operatorsApi = axios.create({
  baseURL: `${URL}${projectsPath}`,
});
// experiments path
const experimentsPath = '/experiments';
// operators path
const operatorsPath = '/operators';

// API METHODS
/**
 * List Operators
 * @param {string} projectId
 * @returns {Promise}
 */
const listOperators = (projectId, experimentId) =>
  new Promise((resolve, reject) => {
    // requesting operators
    operatorsApi
      .get(`/${projectId}${experimentsPath}/${experimentId}${operatorsPath}`)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  });

/**
 * Detail Operator
 * @param {string} projectId
 * @returns {Promise}
 */
/* const detailOperator = (projectId, operatorId) =>
  new Promise((resolve, reject) => {
    // requesting operator
    operatorsApi
      .get(`/${projectId}${operatorsPath}/${operatorId}`)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  }); */

/**
 * Create Operator
 * @param {string} projectId
 * @param {string} operatorName
 * @returns {Promise}
 */
/* const createOperator = (projectId, operatorName) =>
  new Promise((resolve, reject) => {
    // creating body object
    const body = {
      name: operatorName,
    };

    // creating operator
    operatorsApi
      .post(`/${projectId}${operatorsPath}`, body)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  }); */

/**
 * Update Operator
 * @param {string} projectId
 * @param {string} operatorId
 * @param {Object} operator
 * @returns {Promise}
 */
/* const updateOperator = (projectId, operatorId, operator) =>
  new Promise((resolve, reject) => {
    // updating operator
    operatorsApi
      .patch(`/${projectId}${operatorsPath}/${operatorId}`, operator)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  }); */

/**
 * Delete Operator
 * @param {string} projectId
 * @param {string} operatorId
 * @returns {Promise}
 */
/* const deleteOperator = (projectId, operatorId) =>
  new Promise((resolve, reject) => {
    // deleting operator
    operatorsApi
      .delete(`/${projectId}${operatorsPath}/${operatorId}`)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  }); */

// EXPORT DEFAULT
export default {
  listOperators,
  /*   detailOperator,
  createOperator,
  updateOperator,
  deleteOperator, */
};
