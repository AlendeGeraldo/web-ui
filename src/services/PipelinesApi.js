// CORE LIBS
import axios from 'axios';

// CONSTANTS
// api base url
const URL = process.env.REACT_APP_PIPELINES_API || 'http://localhost:3000';
// api object
const pipelinesApi = axios.create({
  baseURL: `${URL}`,
});
// train path
const trainPath = '/train';

/**
 * Train Experiment
 * @param {Object} trainObject
 * @returns {Promise}
 */
const trainExperiment = (trainObject) =>
  new Promise((resolve, reject) => {
    // training experiment
    pipelinesApi
      .post(`${trainPath}`, trainObject)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  });

// EXPORT DEFAULT
export default {
  trainExperiment,
};
