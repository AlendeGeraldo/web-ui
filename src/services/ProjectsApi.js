// CORE LIBS
import axios from 'axios';

// CONSTANTS
// url
const URL = process.env.REACT_APP_PROJECTS_API || 'http://localhost:3000';
// api object
const projectsApi = axios.create({
  baseURL: URL,
});

// API METHODS
/**
 * List Projects
 * @returns {Promise}
 */
const listProjects = () =>
  new Promise((resolve, reject) => {
    // requesting projects
    projectsApi
      .get(`/projects`)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  });

/**
 * Detail Project
 * @param {string} projectId
 * @returns {Promise}
 */
const detailProject = (projectId) =>
  new Promise((resolve, reject) => {
    // requesting project
    projectsApi
      .get(`/projects/${projectId}`)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  });

/**
 * Create Project
 * @param {string} projectName
 * @returns {Promise}
 */
const createProject = (projectName) =>
  new Promise((resolve, reject) => {
    // creating body object
    const body = {
      name: projectName,
    };

    // requesting project
    projectsApi
      .post(`/projects`, body)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  });

/**
 * Update Project
 * @param {string} projectId
 * @param {string} projectName
 * @returns {Promise}
 */
const updateProject = (projectId, projectName) =>
  new Promise((resolve, reject) => {
    // creating body object
    const body = {
      name: projectName,
    };

    // requesting project
    projectsApi
      .patch(`/projects/${projectId}`, body)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  });

export default { listProjects, detailProject, createProject, updateProject };

// TODO: implementar delete project (n√£o encontrei referencia)
