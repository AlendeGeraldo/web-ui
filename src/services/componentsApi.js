/* eslint-disable consistent-return */
import axios from 'axios';
import { message } from 'antd';

export const URL =
  process.env.REACT_APP_PROJECTS_API || 'http://localhost:3000';

export const uploadUrl = `${URL}/components/upload`;

export const downloadUrl = `${URL}/components/download`;

export const componentsApi = axios.create({
  baseURL: URL,
});

export const getAllComponents = async () => {
  try {
    const response = await componentsApi.get(`/components`);
    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const getComponent = async (id) => {
  try {
    const response = await componentsApi.get(`/components/${id}`);

    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const createComponent = async (componentName) => {
  try {
    const body = {
      name: componentName,
    };
    const response = await componentsApi.post(`/components/`, body);

    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const updateComponent = async (id, componentName) => {
  try {
    const body = {
      name: componentName,
    };
    const response = await componentsApi.patch(`/components/${id}`, body);

    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const updateParameters = async (id, componentParameters) => {
  try {
    const body = {
      parameters: componentParameters,
    };
    const response = await componentsApi.patch(`/components/${id}`, body);

    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const deleteFiles = async (id, componentFiles) => {
  try {
    const body = {
      files: componentFiles,
    };
    const response = await componentsApi.post(
      `/components/deleteFiles/${id}`,
      body
    );
    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const deleteComponent = async (id) => {
  try {
    const response = await componentsApi.delete(`/components/${id}`);
    return response;
  } catch (error) {
    message.error(error.message);
  }
};
