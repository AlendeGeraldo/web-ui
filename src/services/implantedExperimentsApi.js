// CORE LIBS
import axios from 'axios';

// CONSTANTS
// api base url
const URL = process.env.REACT_APP_PIPELINES_API || 'http://localhost:3000';
// api object
const pipelinesApi = axios.create({
  baseURL: `${URL}`,
});
const deploymentsPath = '/deployments';

/**
 * Get Deployed Experiments
 * @returns {Promise}
 */
const getDeployedExperiments = () =>
  new Promise((resolve, reject) => {
    // get deployed experiment
    pipelinesApi
      .get(deploymentsPath)
      // success
      .then((response) => resolve(response))
      // error
      .catch((error) => reject(error));
  });

// EXPORT DEFAULT
export default {
  getDeployedExperiments,
};
