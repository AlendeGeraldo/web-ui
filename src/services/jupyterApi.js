/* eslint-disable consistent-return */
import { message } from 'antd';
import axios from 'axios';
import uuidv4 from 'uuid/v4';
import Cookies from 'universal-cookie';

import * as componentsServices from './componentsApi';

export const URL = process.env.REACT_APP_JUPYTER_API || '';

export const componentsApi = axios.create({
  baseURL: URL,
});

export const getNamespaces = async () => {
  try {
    const response = await componentsApi.get(`/jupyter/api/namespaces`);
    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const getNotebook = async (namespace) => {
  try {
    const response = await componentsApi.get(
      `/jupyter/api/namespaces/${namespace}/notebooks`
    );
    return response;
  } catch (error) {
    message.error(error.message);
  }
};

export const uploadFile = async (id, namespace, notebook, fileName) => {
  try {
    const responseFileBase64 = await componentsServices.downloadBase64(
      id,
      fileName
    );

    if (!responseFileBase64) {
      return;
    }

    const { payload } = responseFileBase64.data;
    const body = {
      content: payload,
      format: 'base64',
      name: fileName,
      path: fileName,
      type: 'file',
    };

    const uuid = uuidv4();
    const cookies = new Cookies();
    cookies.set('_xsrf', uuid, { path: '/' });

    const response = await componentsApi.put(
      `/notebook/${namespace}/${notebook}/api/contents/${fileName}`,
      body,
      {
        credentials: 'include',
        headers: {
          'X-XSRFToken': uuid,
        },
      }
    );
    return response;
  } catch (error) {
    message.error(error.message);
  }
};
