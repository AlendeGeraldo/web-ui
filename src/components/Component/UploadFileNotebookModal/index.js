/* eslint-disable react/forbid-prop-types */
/* eslint-disable jsx-a11y/anchor-is-valid */
/* eslint-disable react/destructuring-assignment */
import React from 'react';

import PropTypes from 'prop-types';

import { Form, Modal, message, Select, Spin } from 'antd';

import * as jupyterServices from '../../../services/jupyterApi';

const { Option } = Select;

/* 
  This function is used to check if form has errors
*/
const hasErrors = (fieldsError) => {
  return Object.keys(fieldsError).some((field) => fieldsError[field]);
};

class UploadFileNotebookModal extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      loading: false,
      listNotebook: [],
    };
  }

  sleep = (milliseconds) => {
    return new Promise((resolve) => setTimeout(resolve, milliseconds));
  };

  onCancelModal = () => {
    const { onCancel } = this.props;
    this.setState({ listNotebook: [] });
    onCancel();
  };

  onChangeNamespace = async (namespaceSelected) => {
    this.setState({ loading: true });
    const response = await jupyterServices.getNotebook(namespaceSelected);

    let optionsNotebooks;
    if (response) {
      const { notebooks } = response.data;
      if (notebooks && notebooks.length > 0) {
        optionsNotebooks = notebooks.map((notebook) => (
          <Option key={notebook.name} value={notebook.name}>
            {notebook.name}
          </Option>
        ));
      } else {
        optionsNotebooks = [
          <Option
            key='new-notebook'
            value='new-notebook'
            onClick={() => {
              window.open('/jupyter/');
            }}
          >
            <a>Criar notebook</a>
          </Option>,
        ];
      }
    }
    this.setState({
      listNotebook: optionsNotebooks,
    });
    this.props.form.setFieldsValue({
      notebook: '',
    });
    this.setState({ loading: false });
  };

  handleSubmit = (e) => {
    e.preventDefault();
    this.props.form.validateFields(async (err, values) => {
      if (err) {
        return;
      }

      const { filePath, fileName } = this.props;
      const { namespace, notebook } = values;
      this.setState({ loading: true });

      const response = await jupyterServices.uploadFile(
        namespace,
        notebook,
        filePath,
        fileName
      );

      if (response) {
        message.success(`Upload realizado com sucesso.`);
        this.onCancelModal();
        await this.sleep(1000);
        window.open(`/notebook/${namespace}/${notebook}/notebooks/${fileName}`);
      }

      this.setState({ loading: false });
    });
  };

  render() {
    const { namespaces, visible, form } = this.props;
    const { getFieldDecorator, getFieldsError } = form;
    const { listNotebook } = this.state;

    let optionsNamespaces;
    if (namespaces) {
      optionsNamespaces = namespaces.map((namespace) => (
        <Option key={namespace.namespace} value={namespace.namespace}>
          {namespace.namespace}
        </Option>
      ));
    }

    const formUI = (
      <Form layout='vertical'>
        <Form.Item label='Namespaces: '>
          {getFieldDecorator('namespace', {
            rules: [{ required: true, message: ' ' }],
          })(
            <Select onChange={this.onChangeNamespace}>
              {optionsNamespaces}
            </Select>
          )}
        </Form.Item>
        <Form.Item label='Notebooks: '>
          {getFieldDecorator('notebook', {
            rules: [{ required: true, message: ' ' }],
          })(<Select>{listNotebook}</Select>)}
        </Form.Item>
      </Form>
    );

    return (
      <Modal
        visible={visible}
        title='Editar arquivo no jupyter notebook'
        okText='Editar'
        cancelText='Cancelar'
        onCancel={this.onCancelModal}
        onOk={this.handleSubmit}
        okButtonProps={{ disabled: hasErrors(getFieldsError()) }}
      >
        <Spin spinning={this.state.loading} delay={100}>
          {formUI}
        </Spin>
      </Modal>
    );
  }
}

UploadFileNotebookModal.propTypes = {
  filePath: PropTypes.string.isRequired,
  fileName: PropTypes.string.isRequired,
  namespaces: PropTypes.arrayOf(PropTypes.object).isRequired,
  visible: PropTypes.bool.isRequired,
  onCancel: PropTypes.func.isRequired,
  form: PropTypes.shape({
    getFieldDecorator: PropTypes.func,
    getFieldsError: PropTypes.func,
  }).isRequired,
};

export default Form.create({ name: 'upload-file-notebook' })(
  UploadFileNotebookModal
);
