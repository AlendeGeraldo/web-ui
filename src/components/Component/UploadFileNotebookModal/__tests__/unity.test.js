import React from 'react';
import { act } from 'react-dom/test-utils';
import { shallow, mount } from 'enzyme';
import nock from 'nock';
import { Form, Modal, message, Select, Spin } from 'antd';
import UploadFileNotebookModal from '..';
import * as componentsServices from '../../../../services/componentsApi';
import * as jupyterServices from '../../../../services/jupyterApi';

const namespaces = [
  {
    namespace: 'anonymous',
    role: 'contributor',
    user: 'anonymous@kubeflow.org',
  },
];

const notebooks = [
  {
    name: 'my-notebook-server',
  },
];

const sleep = (milliseconds) => {
  return new Promise((resolve) => setTimeout(resolve, milliseconds));
};

const onCancel = () => {};

describe('UploadFileNotebookModal component', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('is expected render without crashing', () => {
    shallow(
      <UploadFileNotebookModal
        visible
        namespaces={namespaces}
        filePath='filePath'
        fileName='fileName'
        onCancel={onCancel}
      />
    );
  });

  it('is expected render html correctly', () => {
    const wrapper = shallow(
      <UploadFileNotebookModal
        visible
        namespaces={namespaces}
        filePath='filePath'
        fileName='fileName'
        onCancel={onCancel}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render self and subcomponents', () => {
    const wrapper = mount(
      <UploadFileNotebookModal
        visible
        namespaces={namespaces}
        filePath='filePath'
        fileName='fileName'
        onCancel={onCancel}
      />
    );

    expect(wrapper.find(Modal).exists()).toBeTruthy();
    expect(wrapper.find(Spin).exists()).toBeTruthy();
    expect(wrapper.find(Form).exists()).toBeTruthy();
  });

  it('onOk should be called', () => {
    // work around to ignore console log antd
    jest.spyOn(console, 'warn').mockImplementation(() => {});
    jest.spyOn(console, 'error').mockImplementation(() => {});

    const wrapper = mount(
      <UploadFileNotebookModal
        visible
        namespaces={namespaces}
        filePath='filePath'
        fileName='fileName'
        onCancel={onCancel}
      />
    );

    const event = { preventDefault: () => {} };
    const modal = wrapper.find(Modal).instance();
    const spyHandleOk = jest.spyOn(modal, 'handleOk');
    modal.handleOk(event);
    expect(spyHandleOk).toHaveBeenCalled();
  });

  it('onCancel should be called', () => {
    const wrapper = mount(
      <UploadFileNotebookModal
        visible
        namespaces={namespaces}
        filePath='filePath'
        fileName='fileName'
        onCancel={onCancel}
      />
    );
    act(() => {
      const modal = wrapper.find(Modal).instance();
      const spyHandleCancel = jest.spyOn(modal, 'handleCancel');
      modal.handleCancel();
      expect(spyHandleCancel).toHaveBeenCalled();
    });
  });

  it('onChangeNamespace without notebook', async () => {
    const wrapper = mount(
      <UploadFileNotebookModal
        visible
        namespaces={namespaces}
        filePath='filePath'
        fileName='fileName'
        onCancel={onCancel}
      />
    );

    // mock get notebook list response
    nock(jupyterServices.URL)
      .get(`/jupyter/api/namespaces/anonymous/notebooks`)
      .reply(200);

    wrapper
      .find(Select)
      .at(0)
      .simulate('click');
    wrapper
      .find('MenuItem')
      .at(0)
      .simulate('click');
    await sleep(1000);
    wrapper
      .find(Select)
      .at(1)
      .simulate('click');
    expect(
      wrapper
        .find('MenuItem')
        .at(1)
        .props().value
    ).toBe('new-notebook');
  });

  it('onChangeNamespace with notebook', async () => {
    const wrapper = mount(
      <UploadFileNotebookModal
        visible
        namespaces={namespaces}
        filePath='filePath'
        fileName='fileName'
        onCancel={onCancel}
      />
    );

    // mock get notebook list response
    nock(jupyterServices.URL)
      .get(`/jupyter/api/namespaces/anonymous/notebooks`)
      .reply(200, { notebooks });

    wrapper
      .find(Select)
      .at(0)
      .simulate('click');
    wrapper
      .find('MenuItem')
      .at(0)
      .simulate('click');
    await sleep(1000);
    wrapper
      .find(Select)
      .at(1)
      .simulate('click');
    expect(
      wrapper
        .find('MenuItem')
        .at(1)
        .props().value
    ).toBe('my-notebook-server');
  });

  it('create new notebook', async () => {
    const wrapper = mount(
      <UploadFileNotebookModal
        visible
        namespaces={namespaces}
        filePath='filePath'
        fileName='fileName'
        onCancel={onCancel}
      />
    );

    // mock get notebook list response
    nock(jupyterServices.URL)
      .get(`/jupyter/api/namespaces/anonymous/notebooks`)
      .reply(200);

    // spy on windows open
    const spyGlobalOpen = jest.spyOn(global, 'open');

    wrapper
      .find(Select)
      .at(0)
      .simulate('click');
    wrapper
      .find('MenuItem')
      .at(0)
      .simulate('click');
    await sleep(1000);
    wrapper
      .find(Select)
      .at(1)
      .simulate('click');
    wrapper
      .find('MenuItem')
      .at(1)
      .simulate('click');

    expect(spyGlobalOpen).toHaveBeenCalledWith('/jupyter/');
  });

  it('handleSubmit', async () => {
    const wrapper = mount(
      <UploadFileNotebookModal
        visible
        namespaces={namespaces}
        filePath='filePath'
        fileName='fileName'
        onCancel={onCancel}
      />
    );

    // mock get notebook list response
    nock(jupyterServices.URL)
      .get(`/jupyter/api/namespaces/anonymous/notebooks`)
      .reply(200, { notebooks });

    // mock download base 64 response
    nock(componentsServices.URL)
      .post(`/components/downloadBase64`)
      .reply(200, {
        payload: 'payload',
      });

    // mock notebook file upload response
    nock(jupyterServices.URL)
      .put(`/notebook/anonymous/my-notebook-server/api/contents/fileName`)
      .reply(200);

    wrapper
      .find(Select)
      .at(0)
      .simulate('click');
    wrapper
      .find('MenuItem')
      .at(0)
      .simulate('click');
    await sleep(1000);
    wrapper
      .find(Select)
      .at(1)
      .simulate('click');
    wrapper
      .find('MenuItem')
      .at(1)
      .simulate('click');

    const spyMsgSuccess = jest.spyOn(message, 'success');
    const event = { preventDefault: () => {} };
    const modal = wrapper.find(Modal).instance();
    modal.handleOk(event);
    await sleep(2000);
    expect(spyMsgSuccess).toHaveBeenCalledWith(`Upload realizado com sucesso.`);
  });
});
