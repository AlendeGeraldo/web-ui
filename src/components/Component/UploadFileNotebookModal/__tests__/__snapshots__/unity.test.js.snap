// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadFileNotebookModal component is expected render html correctly 1`] = `
<UploadFileNotebookModal
  fileName="fileName"
  filePath="filePath"
  form={
    Object {
      "getFieldDecorator": [Function],
      "getFieldError": [Function],
      "getFieldInstance": [Function],
      "getFieldProps": [Function],
      "getFieldValue": [Function],
      "getFieldsError": [Function],
      "getFieldsValue": [Function],
      "isFieldTouched": [Function],
      "isFieldValidating": [Function],
      "isFieldsTouched": [Function],
      "isFieldsValidating": [Function],
      "isSubmitting": [Function],
      "resetFields": [Function],
      "setFields": [Function],
      "setFieldsInitialValue": [Function],
      "setFieldsValue": [Function],
      "submit": [Function],
      "validateFields": [Function],
      "validateFieldsAndScroll": [Function],
    }
  }
  namespaces={
    Array [
      Object {
        "namespace": "anonymous",
        "role": "contributor",
        "user": "anonymous@kubeflow.org",
      },
    ]
  }
  onCancel={[Function]}
  visible={true}
/>
`;
